### DISCUSSION TITLE
multiple choice schema array of map

### DISCUSSION THREAD

**Discussion by robojiannis** [firecmsco/firecms#357] [Q&A]
Hi there,

i have a collection with a simple array of map schema. Here's a simplified version of it:
```
{ 
dataType: "array",
of: {
  "dataType:"map",
  properties: {
    cards: {
      dataType: "reference",
      path: "cards",   
    },
    amount: {
      dataType: "number",
    }
}
```

With this setup, I can select only one card for each array element. 
Instead I'd like to select multiple cards, which would create the map correctly with a default value for amount. 

Is this possible? I couldn't find something in the documentation.

**Comment by fgatti675** [comment]
I guess you could have an array of references then?
```
{ 
dataType: "array",
of: {
  "dataType:"map",
  properties: {
    cards: {
      dataType: "array",
      of: {
            dataType: "reference",
            path: "cards",   
       }
    },
    amount: {
      dataType: "number",
    }
}
```

**Reply by robojiannis** [reply]
thanks! I tried that, but it doesn't work unfortunately.
The map/result I need is one item mapped to one amount:
```
cards = [
  {amount:1, card: "/cards/02"},
  {amount: 3, card:"card/03"}
]
```

your proposal maps multiple items to one amount:

```
cards = [ 
  {
    amount:1, 
    cards: [
       "/cards/01",
       "/cards/02"
    ]
  },
  {
     amount: 3, 
     cards:[
      "card/03", 
      "/cards/04"
     ]
  }
}]
```
